File: MessagingPrototype/src/main/scala/messaging/Message.scala
package messaging

import java.time.LocalDateTime
import javax.persistence._
import org.hibernate.annotations.Type


/**
 * Created by justusadam on 18/11/14.
 *
 * A String message that can be sent.
 * Generally the "sent" date should not be set manually
 */
@Entity
class Message{

  private var message:String = _
  private var sender:Integer = _
  private var recipient:Integer = _

  @Type(`type`="org.jadira.usertype.dateandtime.threeten.PersistentLocalDateTime")
  var sent:LocalDateTime = LocalDateTime.now()

  @Type(`type`="org.jadira.usertype.dateandtime.threeten.PersistentLocalDateTime")
  var dateReceived:LocalDateTime = _
  var read = false

  @GeneratedValue
  @Id
  var id:Int = _

  def this(message:String, sender:Integer, recipient:Integer) {
    this()
    this.sender = sender
    this.recipient = recipient
    this.message = message
  }

  def getMessage = message

  def setMessage(message:String) = this.message = message

  def getSender = sender

  def setSender(sender:Integer) = this.sender = sender

  def getRecipient = recipient

  def setRecipient(recipient:Integer) = this.recipient = recipient

  def getDateReceived = dateReceived

  def received:Boolean = {
    read
  }

  override def toString = {
    sender + " [" + sent.toString + "]" + ": " + message
  }

  def readMessage = {
    read = true

    message
  }
}

File: src/main/java/messaging/Message.java
package messaging;

import org.hibernate.annotations.Type;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.time.LocalDateTime;

/**
 * Created by justusadam on 09/12/14.
 */
@Entity
public class Message {

    @GeneratedValue
    @Id
    private long id;

    private String message;

    @Type(type = "org.jadira.usertype.dateandtime.threeten.PersistentLocalDateTime")
    private LocalDateTime dateSent;

    @Type(type = "org.jadira.usertype.dateandtime.threeten.PersistentLocalDateTime")
    private LocalDateTime dateReceived;

    private long sender;

    private long receiver;

    private boolean read = false;

    @Deprecated
    public Message(){}

    public Message(String message, LocalDateTime dateSent, long sender, long receiver) {
        this.message = message;
        this.dateSent = dateSent;
        this.sender = sender;
        this.receiver = receiver;
    }

    public Message(String message, long sender, long receiver) {
        this.message = message;
        this.sender = sender;
        this.receiver = receiver;
        this.dateSent = LocalDateTime.now();
    }

    public LocalDateTime getDateSent() {
        return dateSent;
    }

    public LocalDateTime getDateReceived() {
        return dateReceived;
    }

    public void setDateReceived(LocalDateTime dateReceived) {
        this.dateReceived = dateReceived;
    }

    public long getSender() {
        return sender;
    }

    public long getReceiver() {
        return receiver;
    }

    public long getId() {
        return id;
    }

    public String readMessage(){
        this.read = true;
        return this.message;
    }

    public boolean isRead() {
        return read;
    }

    @Override
    public String toString(){
        return sender + "  [" + dateReceived.toString() + "]  " + message;
    }
}
